src/sinks/appsignal/mod.rs

```diff
@@ -0,0 +1,299 @@
+//! The AppSignal sink
+//!
+//! This sink provides downstream support for `AppSignal` to collect logs and a subset of Vector
+//! metric types. These events are sent to the `appsignal-endpoint.net` domain, which is part of
+//! the `appsignal.com` infrastructure.
+//!
+//! Logs and metrics are stored on an per app basis and require an app-level Push API key.
+
+#[cfg(all(test, feature = "appsignal-integration-tests"))]
+mod integration_tests;
+
+use bytes::Bytes;
+use futures::{FutureExt, SinkExt};
+use http::{header::AUTHORIZATION, Request, Uri};
+use hyper::Body;
+use serde_json::json;
+use snafu::{ResultExt, Snafu};
+use vector_common::sensitive_string::SensitiveString;
+use vector_config::configurable_component;
+
+use crate::{
+    codecs::Transformer,
+    config::{AcknowledgementsConfig, DataType, Input, SinkConfig, SinkContext},
+    event::Event,
+    http::HttpClient,
+    sinks::{
+        util::{
+            encoding::write_all,
+            http::{BatchedHttpSink, HttpEventEncoder, HttpSink},
+            BatchConfig, BoxedRawValue, Compression, Compressor, JsonArrayBuffer,
+            SinkBatchSettings, TowerRequestConfig,
+        },
+        BuildError,
+    },
+    tls::TlsSettings,
+};
+
+#[derive(Debug, Snafu)]
+enum FinishError {
+    #[snafu(display(
+        "Failure occurred during writing to or finalizing the compressor: {}",
+        source
+    ))]
+    CompressionFailed { source: std::io::Error },
+}
+
+/// Supported compression types for the AppSignal sink.
+#[configurable_component]
+#[derive(Copy, Clone, Debug, Derivative, Eq, PartialEq)]
+#[serde(rename_all = "lowercase")]
+#[derivative(Default)]
+enum AppsignalCompression {
```

Comment:
👋 I've decided I'm less concerned about `gzip` by default, but would like to see a `none` option here.